### PR TITLE
Bump the typescript-eslint group 

### PR DESCRIPTION
**Pull request by hsubramanianaks**
This PR supersedes https://github.com/Azure/vscode-aks-tools/pull/833 and fixes the eslint errors while running in ubuntu/linux. 

Here is failing run - https://github.com/Azure/vscode-aks-tools/actions/runs/10256769131/job/28378026288?pr=833

image

**Issue#1: error  'e' is defined but never used  @typescript-eslint/no-unused-vars**

most of the places I removed them or included the unused var in error statements. 

**Issue#2: error  An empty interface declaration allows any non-nullish value, including literals like `0` and `""`.**

changed the empty interface to type object. ex: export interface InitialState {} --> export type InitialState = object;

### CODE REVIEW

File: src/commands/draft/draftCommands.ts [hunk 1]
```diff
@@ -1,33 +1,32 @@
...
+    FileType,
+    Uri,
     window,
     workspace,
-    AuthenticationSession,
-    Uri,
-    FileType,
+    WorkspaceFolder,
 } from "vscode";
-import { DraftDockerfileDataProvider, DraftDockerfilePanel } from "../../panels/draft/DraftDockerfilePanel";
```
Comment (ReinierCC) [inline src/commands/draft/draftCommands.ts:12]: Is there a best practice in regard to the ordering of import statements?

File: src/auth/azureAuth.ts [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
     ProgressOptions,
     QuickPickItem,
+    Disposable as VsCodeDisposable,
     window,
 } from "vscode";
-import { AzureSessionProvider, ReadyAzureSessionProvider, Tenant, TokenInfo, isReady } from "./types";
-import { Environment } from "@azure/ms-rest-azure-env";
 import { getConfiguredAzureEnv } from "../commands/utils/config";
 import { Errorable, failed } from "../commands/utils/errorable";
-import { TokenCredential } from "@azure/core-auth";
```
Comment (Tatsinnit) [inline src/auth/azureAuth.ts:13]: 💡 Thoughts please: it almost looks like you have used some formatter for all these formatting changes? do we need all this? for the dependabot fix PR, jsut thinking with scope of keep this all contained.

File: src/webview-contract/webviewDefinitions/inspektorGadget.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { WebviewDefinition } from "../webviewTypes";
 
-export interface InitialState {}
+export type InitialState = object;
```
Comment (Tatsinnit) [inline src/webview-contract/webviewDefinitions/inspektorGadget.ts:3]: 💡 Idea:  Just something to take a quick peak, https://github.com/Azure/vscode-aks-tools/compare/main...Tatsinnit:vscode-aks-tools:fix/dependbot-packages-eslint if you want to port any of these.

File: src/auth/azureAuth.ts (hunk 2 above)
Comment (hsubramanianaks) [reply src/auth/azureAuth.ts:13]: I used normal vscode to do my changes not sure what changed w.r.t formatter, will recheck on it. Thanks for catching it.

**Comment by hsubramanianaks** [comment]
closing this since it is taken care in https://github.com/Azure/vscode-aks-tools/pull/845

**Pull request closed without merging**